vortex-scalar/src/arbitrary.rs

```diff
@@ -82,19 +84,12 @@ fn random_pvalue(u: &mut Unstructured, ptype: &PType) -> Result<PValue> {
 /// Generate an arbitrary decimal scalar confined to the given bounds of precision and scale.
 pub fn random_decimal(u: &mut Unstructured, decimal_type: &DecimalDType) -> Result<ScalarValue> {
     let precision = decimal_type.precision();
-    if precision <= i128::MAX_PRECISION {
-        Ok(ScalarValue(InnerScalarValue::Decimal(DecimalValue::I128(
-            u.int_in_range(
-                i128::MIN_BY_PRECISION[precision as usize]
-                    ..=i128::MAX_BY_PRECISION[precision as usize],
-            )?,
-        ))))
-    } else {
-        Ok(ScalarValue(InnerScalarValue::Decimal(DecimalValue::I256(
-            u.int_in_range(
-                i256::MIN_BY_PRECISION[precision as usize]
-                    ..=i256::MAX_BY_PRECISION[precision as usize],
-            )?,
-        ))))
-    }
+    let value = match_each_decimal_value_type!(smallest_decimal_value_type(decimal_type), |D| {
```

Comment:
ah yes, this code predates other decimal types

Reply:
I think the actual problem was the off-by-one in the Precision array.